libdnf/utils/CompressedFile.hpp

```diff
@@ -0,0 +1,27 @@
+#ifndef LIBDNF_GZFILE_HPP
+#define LIBDNF_GZFILE_HPP
+
+#include "File.hpp"
+
+namespace libdnf {
+
+class CompressedFile : public libdnf::File
```

Comment:
If I delete instantion of the class who close file? Where is destructor?

Reply:
`void close() override;`. I do not automatically open the file, if you open it, you should close it. So it is not the problem of class.

Reply:
It is problem of class. And big problem.
Rule is that owner frees its resources during destruction. We do not want resource (memory, filedescriptor, ...) leaks.
Example:
The std::fstream (C++ standard class) automaticaly closes file during destruction.

And second big problem is that you are inconsistent.
The class File (your class) closes file during destruction but its derived class CompressedFile do not. So we can get container with File objects and some of them automatically close and some of them (upcasted CompresedFiles) do not.

Third problem. The class File contains virtual methods. What about virtual destructor?